pkg/network/state_test.go

```diff
@@ -2590,6 +2590,158 @@ func TestDNSPIDCollision(t *testing.T) {
 	assert.Empty(t, delta.Conns[1].DNSStats, "dns stats should not be empty")
 }
 
+// Helper function to create a unique ConnectionStats
+func createTestConnectionStats(cookie StatCookie) *ConnectionStats {
+	return &ConnectionStats{
+		ConnectionTuple: ConnectionTuple{
+			Source: util.AddressFromString("1.1.1.1"),
+			Dest:   util.AddressFromString("2.2.2.2"),
+			SPort:  1234,
+			DPort:  80,
+			Type:   TCP,
+		},
+		Cookie: cookie,
+		Monotonic: StatCounters{
+			SentBytes: 100,
+			RecvBytes: 50,
+		},
+		LastUpdateEpoch: uint64(time.Now().UnixNano()),
+	}
+}
+
+func TestNearCapacityFlagSetAndCheck(t *testing.T) {
+	// Use a small size for easier testing
+	const maxClosedConns = 10
+	const closedConnsThresholdRatio = 0.9
+	const capacityThreshold = uint32(float64(maxClosedConns) * closedConnsThresholdRatio)
+
+	ns := NewState(nil, 1*time.Minute, maxClosedConns, closedConnsThresholdRatio, 100, 100, 100, 100, 100, 100, false, false).(*networkState)
+	require.NotNil(t, ns)
+
+	clientID := "test-client-flag-set"
+	ns.RegisterClient(clientID)
+
+	// Add connections below threshold
+	ns.Lock()
+	for i := 0; uint32(i) < capacityThreshold-1; i++ { // Add 8 connections
+		ns.storeClosedConnection(createTestConnectionStats(StatCookie(i)))
```

Comment:
You can use `networkState.StoreClosedConnection` in the tests which acquires the lock.